iree/turbine/kernel/wave/hoisting.py

```diff
@@ -94,8 +99,8 @@ def hoist_loop_invariant_ops(trace: CapturedTrace, constraints: list[Constraint]
                         hoistable_op.replace_all_uses_with(new_op)
                         hoistable_op.erase()
                         if isinstance(hoistable_op, Read):
-                            capture_arg = captured_vars.index(hoistable_op.memory)
-                            new_op.update_arg("memory", implicit_captures[capture_arg])
+                            root_var = hoistable_op.memory.meta["lifted"]
```

Comment:
For a separate PR, can we add a get_lifted or get_captured_var function?

Reply:
SG, will put in separate PR :)